clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus_test.go

```diff
@@ -85,12 +85,26 @@ type summaryEntry struct {
 }
 
 type fakeQueryExecutor struct {
-	samples []*sample
+	metricSamples map[string][]*sample
 }
 
 func (ex *fakeQueryExecutor) Query(query string, _ time.Time) ([]*model.Sample, error) {
+
+	var foundSamples bool
+	var metricSamples []*sample
+	for k, v := range ex.metricSamples {
```

Comment:
I don't understand it - why you don't want to have:
```
metricSamples, ok := ex.metricsSamples[query]
if !ok {
  return nil, fmt.Errorf(...)
}
...
```

Reply:
it is a simplification - allows for defining a simpler query for the tests.

for example, a test definition can define output for `apiserver_request_sli_latency` query whereas the actual query used by the cl is: `quantile_over_time(0.99, apiserver:apiserver_request_sli_latency_1m:histogram_quantile{verb!="WATCH", subresource!~"log|exec|portforward|attach|proxy"}[60s])`

Reply:
OK - I think it's debatable what is easier to follow, but.. I'm not going to fight for it :)